install/kubernetes/cilium/templates/clustermesh-apiserver/deployment.yaml

```diff
@@ -167,6 +174,11 @@ spec:
         - name: etcd-admin-client
           mountPath: /var/lib/cilium/etcd-secrets
           readOnly: true
+        {{- if ne .Values.clustermesh.apiserver.tls.authMode "legacy" }}
```

Comment:
Any reason not to always mount this?

Reply:
(The configMap Volume could just be `optional: true`)

Reply:
No particular reasons, it just seemed to me a bit cleaner to not mount it if not needed, as a restart of the pod would still be caused upon authMode change due to the different parameters. I'm open to change it if you feel always mounting it is better.